source/kraken/apply_disruption.cpp

```diff
@@ -190,11 +202,89 @@ struct add_impacts_visitor : public apply_impacts_visitor {
     }
 
     void operator()(nt::StopPoint* stop_point) {
-        LOG4CPLUS_INFO(log, "Disruption on stop point:" << stop_point->uri << " is not handled");
+
+        using namespace boost::posix_time;
+        using namespace boost::gregorian;
+
+        type::ValidityPattern impact_vp{meta.production_date.begin()}; // bitset are all initialised to 0
+        for (const auto& period: impact->application_periods) {
+            auto one_day = boost::posix_time::time_duration{24, 0, 0};
+            for (time_iterator it(period.begin() - one_day, one_day); it < period.end(); ++it) {
+                if (! meta.production_date.contains((*it).date())) { continue; }
+                auto day = ((*it).date() - meta.production_date.begin()).days();
+                impact_vp.add(day);
+            }
+        }
+
+        std::vector<std::pair<const nt::VehicleJourney*, nt::ValidityPattern>> vj_vp_pairs;
+        for (const auto* vj: pt_data.vehicle_journeys) {
+
+            if ((vj->validity_patterns[rt_level]->days & impact_vp.days).none()) {
+                continue;
+            }
+
+            nt::ValidityPattern new_vp{vj->validity_patterns[rt_level]->beginning_date};
+
+            for(const auto& period : impact->application_periods) {
+                new_vp.days |= vj->get_vp_of_sp(*stop_point, rt_level, period).days;
+            }
+
+            if(new_vp.days.none()){
+                // The vj doesn't stop at the impacted stop_point during all the given impact periods
+                continue;
+            }
+            new_vp.days >>= vj->shift;
+            vj_vp_pairs.emplace_back(vj, new_vp);
+        }
+
+        for (auto& vj_vp : vj_vp_pairs) {
+            std::vector<nt::StopTime> new_stop_times;
+            const auto* vj = vj_vp.first;
+            auto& new_vp = vj_vp.second;
+
+            for (const auto& st : vj->stop_time_list) {
+                if (st.stop_point == stop_point) {
+                    continue;
+                }
+                new_stop_times.emplace_back(
+                        st.arrival_time + ndtu::SECONDS_PER_DAY * vj->shift,
+                        st.departure_time + ndtu::SECONDS_PER_DAY * vj->shift,
+                        st.stop_point);
+            }
+
+            auto mvj = vj->meta_vj;
+            mvj->impacted_by.push_back(impact);
+
+            auto nb_rt_vj = mvj->get_vjs_at(rt_level).size();
+            std::string new_vj_uri = mvj->uri + ":" +
+                    type::get_string_from_rt_level(rt_level) + ":" +
+                    std::to_string(nb_rt_vj) + concatenate_impact_uris(*mvj);
+
+            new_vp.days = new_vp.days & (vj->validity_patterns[rt_level]->days >> vj->shift);
+
+            auto* new_vj = mvj->create_discrete_vj(new_vj_uri,
```

Comment:
Don't forget to copy name, company, vehicle_journey_type, odt_message, _vehicle_properties